### PR TITLE
Migrate to docusaurus 2

### LINKED ISSUE #525: [Feature Request] Converting document to Docsaurus 2
**Do you want to request a *feature* or report a *bug*?**

This is a feature request

## Overview

I saw this Issue https://github.com/facebook/docusaurus/issues/1834 that was migrating many of the fb document to docsaurus 2-alpha.

Would it be better for metro to have its document in docsaurus 2?
If so, I'd like to help migrating the document, but I just wanted to make sure if that is ok before I start working on it.

**What is the current behavior?**

Metro document is made with docsaurus 1

**What is the expected behavior?**

write Metro document in docsaurus 2 (alpha)

### PR DESCRIPTION
**Pull request by Naturalclar**
**Summary**

Part of https://github.com/facebook/docusaurus/issues/1834
Closes #525 

Migrated documentation of metro to `docusaurus 2`

Website can be previewed here https://metro-preview.naturalclar.now.sh/

In order for the publish ci to pass, `GIT_USER` needs to be set to someone with publish access in the ci config.
https://v2.docusaurus.io/docs/deployment

**Test plan**
cd website
yarn start

cc: @yangshun

### CODE REVIEW

File: website/src/css/custom.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/* You can override the default Infima variables here. */
+:root {
+  --ifm-color-primary: #ef4242;
+  --ifm-color-primary-dark: #ed2626;
+  --ifm-color-primary-darker: #eb1818;
+  --ifm-color-primary-darkest: #c51111;
+  --ifm-color-primary-light: #f15e5e;
+  --ifm-color-primary-lighter: #f36c6c;
+  --ifm-color-primary-lightest: #f69696;
+}
```
Comment (imbsky) [inline website/src/css/custom.css]: The current default code font size is really small.
```suggestion
:root {
  --ifm-color-primary: #ef4242;
  --ifm-color-primary-dark: #ed2626;
  --ifm-color-primary-darker: #eb1818;
  --ifm-color-primary-darkest: #c51111;
  --ifm-color-primary-light: #f15e5e;
  --ifm-color-primary-lighter: #f36c6c;
  --ifm-color-primary-lightest: #f69696;
  --ifm-code-font-size: 95%;
}
```

File: website/package.json [hunk 2]
```diff
@@ -3,21 +3,24 @@
...
-    "rename-version": "docusaurus-rename-version"
+    "start": "docusaurus start",
+    "build": "docusaurus build",
+    "swizzle": "docusaurus swizzle",
+    "deploy": "docusaurus deploy"
   },
   "dependencies": {
-    "docusaurus": "1.3.2"
+    "@docusaurus/core": "2.0.0-alpha.40",
+    "@docusaurus/preset-classic":"2.0.0-alpha.40",
```
Comment (imbsky) [inline website/package.json]: We can manually rewrite this to use the latest version `2.0.0-alpha.43`.

File: .gitignore [hunk 3]
```diff
@@ -12,3 +12,6 @@ website/node_modules
 website/translated_docs
 .idea
 .vscode
+.docusaurus
+.cache-loader
+yarn-error.log
```
Comment (imbsky) [inline .gitignore]: Insert a final newline here.

File: circle.yml [hunk 4]
```diff
@@ -40,17 +40,14 @@ commands:
...
-          command: publish-gh-pages
+          command: deploy
           working_directory: website
 
 version: 2.1
 jobs:
   run-js-checks:
     docker:
-      - image: circleci/node:8
+      - image: circleci/node:10
```
Comment (imbsky) [inline circle.yml]: I think we should use 12 instead of 10. or did you select 10 for some reason?

File: circle.yml [hunk 5]
```diff
@@ -66,18 +63,9 @@ jobs:
...
-    steps:
-      - checkout
-      - yarn_install
-      - yarn_run:
-          command: jest
-
   publish-to-npm:
     docker:
-      - image: circleci/node:8
+      - image: circleci/node:10
```
Comment (imbsky) [inline circle.yml]: Same as above

File: circle.yml [hunk 6]
```diff
@@ -87,7 +75,7 @@ jobs:
 
   test-and-deploy-website:
     docker:
-      - image: circleci/node:8
+      - image: circleci/node:10
```
Comment (imbsky) [inline circle.yml]: Same as above

**Comment by imbsky** [comment]
> In order for the publish ci to pass, GIT_USER needs to be set to someone with publish access in the ci config.
> https://v2.docusaurus.io/docs/deployment

I think we can bind `CIRCLE_USERNAME` to `GIT_USER`.
https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables

**Comment by imbsky** [comment]
We are currently using `metro-bot`, so we just need to set `metro-bot`.
```yml
deploy_website:
  steps:
    - run: git config --global user.email "metro-bot@users.noreply.github.com"
    - run: git config --global user.name "Website Deployment Script"
    - run: echo "machine github.com login metro-bot password $GITHUB_TOKEN" > ~/.netrc
    - yarn_install:
        working_directory: website
    - yarn_run:
        command: deploy
        working_directory: website
        environment:
          GIT_USER: metro-bot
```

**Comment by imbsky** [comment]
Minor changes are necessary, but overall it looks pretty good! Everything is definitely better than the current one and I love these changes!

**Comment by Naturalclar** [comment]
@imbsky thanks for the review! I've fixed some changes that were pointed out.
As for the CI and environment related stuff, it's a bit out of scope for this PR and I think it's better for the repo owner to decide.

**Comment by imbsky** [comment]
Looks good overall! Indeed, I will leave CI things to the Metro team!

**Comment by Naturalclar** [comment]
@cpojer I'd like to work on the react-native website once all the component updates in the docs are settled :D 
I'm also waiting for the docusaurus 2 to support versioning in the document, as that is pretty crucial for the RN document.
I'm not sure if I can do much about the CI stuff with the deployment (in terms of access rights), so I'd like to leave that to the FB OSS team.

**Comment by yangshun** [comment]
D2 has versioning now but I'd like to hold off on migrating existing mature D1 websites which use versioning. The versioning approach is a bit different and it'd be a pain to migrate for now.

**Comment by JoelMarcey** [comment]
Are you running into permission issues on the actual website push? Normally `metro-bot` does the push of the site, right? If so, maybe our internal tooling didn't renew the permissions correctly.

**Comment by JoelMarcey** [comment]
`metro-bot` looks to have the right permissions from our side. 

@imbsky @Naturalclar Or are you just asking if we are good with the suggested `.yml` changes for CircleCI? If so, I think it looks good.

**Comment by Naturalclar** [comment]
@JoelMarcey got it! I thought I wasn't able to change the environment var, but I just had incorrect syntax for the config, haha

**Comment by imbsky** [comment]
Looks good!

**Comment by yangshun** [comment]
@cpojer what else is blocking us from deploying the new site?

**Pull request closed without merging**